### PR TITLE
Fix SpotBugs issues + Fix the default Java versioncapturing in PluginCompatTesterConfig#getTestJavaVersion() when other options are disabled

### PR DESCRIPTION
**Pull request by basil**
Fixes all SpotBugs issues and enables SpotBugs by default. Each commit fixes exactly one SpotBugs issue and should be self-explanatory. Let me know if I can explain any commit further or if any cleanup is undesired.

### CODE REVIEW

File: plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java [hunk 1]
```diff
@@ -346,7 +346,7 @@ public String getTestJavaVersion() throws IOException {
         final String javaVersionOutput2 = IOUtils.toString(process2.getInputStream());
         // Expected format is something like openjdk full version "1.8.0_181-8u181-b13-2~deb9u1-b13"
         // We shorten it by removing the "full version" in the middle
-        return javaVersionOutput.
+        return javaVersionOutput2.
```
Comment (basil) [inline plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java]: This was a simple bug; we were using the wrong variable here.

File: plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 2]
```diff
@@ -616,9 +615,6 @@ private boolean onlyOnePluginIncluded() {
         top.put("id", DEFAULT_SOURCE_ID);
         JSONObject plugins = new JSONObject();
         try (JarFile jf = new JarFile(war)) {
-            if (pluginGroupIds == null) {
-                pluginGroupIds = new HashMap<>();
-            }
```
Comment (basil) [inline plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java]: This was dead code. We were always passing in a non-`null` value for `pluginGroupIds`.

**Comment by basil** [comment]
CC: @jenkinsci/hacktoberfest

File: plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PluginCompatReport.java [hunk 3]
```diff
@@ -83,8 +83,9 @@ public void save(File reportPath) throws IOException {
         }
 
         // Writing to a temporary report file ...
-        File tempReportPath = new File(reportPath.getAbsolutePath()+".tmp");
-        try (Writer out = new FileWriter(tempReportPath)) {
+        File tempReportPath = new File(reportPath.getAbsolutePath() + ".tmp");
+        try (Writer out =
+                Files.newBufferedWriter(tempReportPath.toPath(), Charset.defaultCharset())) {
```
Comment (oleg-nenashev) [inline plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PluginCompatReport.java]: There is an explicit encoding set in the line below. Should it use that one?

File: plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java (hunk 1 above)
Comment (oleg-nenashev) [reply plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java]: agreed

File: plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java (hunk 2 above)
Comment (oleg-nenashev) [reply plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java]: I suggest adding annotation to the argument to indicate that

File: plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/MultiParentCompileHook.java [hunk 4]
```diff
@@ -28,6 +29,7 @@ public MultiParentCompileHook() {
 
 
     @Override
+    @SuppressFBWarnings(value = "REC_CATCH_EXCEPTION", justification = "silly rule")
```
Comment (oleg-nenashev) [inline plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/MultiParentCompileHook.java]: not informative. Is it for `} catch (Exception e) {` ?

Comment (basil) [reply plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/MultiParentCompileHook.java]: I was just following [existing](https://github.com/jenkinsci/workflow-api-plugin/blob/907cf64feb2f38e93aebbedf87946b0235c3dc93/src/main/java/org/jenkinsci/plugins/workflow/log/LogStorage.java#L125) [precedent](https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/00db760fa039f039cc63b5577a2e7b9f9d700981/src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java#L558) in the Jenkins codebase.

File: plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PluginCompatReport.java (hunk 3 above)
Comment (basil) [reply plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PluginCompatReport.java]: Maybe, but in any case that is a pre-existing issue that wasn't introduced by my PR.

**Comment by oleg-nenashev** [comment]
Sorry, missed that

**Pull request merged**